src/queries/makeQueries.ts

```diff
@@ -83,6 +85,26 @@ Example:
   return waitForOptions;
 }
 
+function getWrongHostComponentNamesError(): string {
+  const reactNativeHostComponentNames = getReactNativeHostComponentNames();
+  const configHostComponentNames = getConfig().hostComponentNames;
+  if (
+    reactNativeHostComponentNames.text !== configHostComponentNames.text ||
+    reactNativeHostComponentNames.textInput !==
+      configHostComponentNames.textInput
+  ) {
+    return `
+
+The wrong host component names are used. This can happen if you use a version of react native that is not compatible with your version of @testing-library/react-native
+
+To fix this, either upgrade to a version that supports your version of react native or add the following line in a setup file :
```

Comment:
```suggestion
To fix this, either upgrade to a version that supports your version of react native or add the following line in a setup file:
```

Reply:
This has been updated with maciej proposal